### DISCUSSION TITLE
Situações constrangedoras encontradas em entrevistas e projetos

### DISCUSSION THREAD

**Discussion by rubenmarcus** [frontendbr/forum#58] [Discussão]
Galera to criando essa issue ,mas como desabafo mesmo.
Até pensei em fazer uma "blacklist" de clientes, porquê com certeza recrutadores e chefes fazem de funcionários. Acho meio anti-etico mas ok.

Era só um desabafo, coisas do tipo:

Experiencias minhas:
"Se precisar manjo de design tambem" _(Isso não quer dizer que eu vou ter que ficar criando cada tela e adaptando, tudo que você me pedir, sou pago pra codificar, não pra criar o tempo todo)_

"Manjo um pouco de infra, de bd e em back-end sou junior" _( Isso nao quer dizer que eu tenha que ter dominio e desenvolver o sistema inteiro e ganhar o salario de um estagiario )_

quando o cliente fala:

"é SÓ ISSO " É muito facil. Não sei como você nao consegue em 2h

etc etc

**Comment by kazzkiq** [comment]
> "Se precisar manjo de design tambem"

Como acabar com a carreira de front-end na empresa em um passo. :laughing: 

Momento desabafo também:

Uma vez eu fiquei de "designer" no job por um tempo (enquanto não cobriam a vaga do cara que tinha saído). Me botaram até pra fazer protótipo navegável em Axure (se você não sabe o que é isso, fique feliz). Quando contrataram outro designer, ao invés de me voltar para o front-end (que tava com pouca demanda) acabaram me deixando como "prototipador" e colocando o novo cara pra fazer o design final dos protótipos aprovados. Pensa numa depressão. Numa semana dev front-end, na outra criador de quadradinhos cinzas em telas-protótipo.

O pior é que a empresa (que não era agência) não estava muito bem das pernas e começou a pegar projetos só de design, mesmo com 90% da mão de obra sendo desenvolvedores. Imagina a felicidade da galera...

... mas pelo menos eles não tavam prototipando. =/

Depois desse (e alguns outros) acontecimentos eu tomei pra mim que se você mexe com ambas as áreas (design e desenvolvimento), pense muito bem antes de explicitar isso para o seu chefe/entrevistador/whatever. As chances de ele te botar pra fazer de tudo um pouco pagando um salário não-compatível são beem grandes, pelo menos nas minhas azaradas experiências.

**Comment by vitorbritto** [comment]
@rubenmarcus tomei a liberdade de modificar o título para algo mais genérico. =]

**Comment by vitorbritto** [comment]
Esse tipo de situação sempre acontece, mesmo que o profissional seja correto, ético e cumpra com suas obrigações no projeto. 

Por mais que algo constrangedor ou chato aconteça, e/ou que me prejudique, eu prefiro guardar comigo. Sou discreto e gosto de ser discreto quanto a isso. Nunca se sabe o que pode acontecer mais a frente e prefiro não fechar portas, nem mesmo criar atritos.

Não a nada de errado em desabafar, mas não costumo falar sem pensar ou publicar coisas do tipo em redes sociais. Não quero aqui dizer, que esteja errado quem o faz. Afinal, é uma ação natural do ser humano. Eu não o faço, pois não me sinto bem. Bola pra frente...

**Comment by felipefialho** [comment]
Interessante esse depoimento de vocês sobre especificar que também "faz Design", é de se imaginar que esse tipo de situação aconteça, já que você se torna uma espécie de coringa.

---

Em entrevista lembro quando ainda estava começando, meados de 2007/2008 (acho que não era nem maior de idade), estava procurando um estágio para ingressar na área (também não tinha certeza do que faria dentro dos mundos web). 

E fiz uma entrevista na Vila Olimpia aqui em São Paulo em uma empresa que ficava em um puta prédio foda mas que tinha só 2 funcionários. Um era o dono o outro um estagiário. O dono estava procurando mais um estagiário. 

O que me impressionou nessa entrevista foi o quanto o dono dessa empresa era arrogante, ele falava de um jeito como se fosse um enviado dos céus para espalhar ouro aqui na terra. A postura arrogante dele continuava com o outro estagiário. 

O trabalho era literalmente para fazer tudo (Back, Front, UX e Design), e bem, contratar dois estagiários para fazer serviço de várias pessoas diz muito sobre a empresa.

---

No trabalho, a pior situação foi logo que estava começando (devia ter uns 6 meses de profissão, meados de 2009), trabalhava em uma fábrica de software que prestava serviço para uma agencia grande, nós transformávamos os layouts deles em projetos. 

Nessa agencia era "normal" trabalhar de madrugada ou dois dias seguidos, e como tínhamos que entregar os projetos, isso acabou se estendendo pra gente. Mas essa não é a questão.

Um dos dias que passei a madrugada trabalhando em um layout sem noção (aqueles típicos de agencia em que o Designer não entende de código) estava tentando fazer a entrega pela manhã para enfim ir pra casa.

Eis que recebo uma ligação, era o Gerente de Projetos dessa agencia, querendo "alinhar" as coisas. Não estava muito gentil no telefone, segundo ele, o layout estava "muito diferente" do PSD. Me perguntou como que eu fazia a verificação do layout e expliquei, que era impossível deixar totalmente fiel ao PSD, e ai...

"Aqui na XXXXXX fazemos o seguinte, a gente tira uma print do browser e coloca em cima do PSD e vai alinhando até tudo estar 100%!"

Ainda tentei argumentar que o PSD enviado estava inconsistente quanto aos espaçamentos, mas não teve jeito. Acreditem ou não, realmente tive que fazer esse trabalho bizarro, e mais... colocando alinhamentos específicos para cada página e não para os elementos para ficar "fiel" ao PSD.

Fico imaginando, se na era de Mobile First, eles continuam colocando prints em cima do PSD :)

**Comment by rubenmarcus** [comment]
@LFeh  sim. eu to numa produtora digital muito parecido com o que você faz fazendo um site de uma convenção pro Itau, que vai ser visto no IE7 e IE8.

Mesmo ritmo de trabalho ,e o designer fez questão de deixar os margins dos inputs de cadastro e etc com valores diferentes.. só nisso eu perdi um tempão pq não batia os alinhamentos de regua na esq. e direita.

Alem de mudança de escopo toda hora, e os prazos ao invés de extender diminuem.

Ai você é o taxado de "incompetente"

**Comment by rubenmarcus** [comment]
O pior que eu não estou ganhando nada pelas horas extras.

**Comment by tcelestino** [comment]
@LFeh, trabalhei em uma agência que um dos sócios, que acreditava ser designer, media com régua e até com papel A4.

**Comment by felipefialho** [comment]
@tcelestino Ahh, se ta zoando...? hahaha

**Comment by rickbenetti** [comment]
@LFeh em 2008 quando trabalhei na Agência Click fui pra um projeto da Meio&&Mensagem os caras dos testes usavam régua, o pessoal do comercial pra insistir que vc estava errado usava régua.

Mas o pior foi com o pessoal da Accenture de testes num projeto da Oi com o @diegoeis ele deve lembrar muito do testes dos 1000cliques por hora, que consistia de os caras clicarem num select que nem locos pra provar que o select customizado dava pau.

**Comment by felipefialho** [comment]
Carario, e eu achando ruim o negócio da print em cima do PSD, nunca mais reclamo! haha

**Comment by mksanti** [comment]
Depois de tudo isso, vejo que algumas madrugas e horas extras não são tão ruins.

**Comment by fernandoguedes** [comment]
Tópico interessante, é importante como profissional se opôr e saber contra argumentar os absurdos que acontecem. De todos, pra mim o mais alarmante é o "só isso", isso desfaz amizades e quebram equipes. Por isso reforço que a importância de demonstrar o como e o porquê da coisa ser feita daquela maneira, e que frutos podem ser colhidos com a entrega de um determinado produto, funcionalidade.

**Comment by nighto** [comment]
Nunca passei pela situação de não receber por horas extras. Já trabalhei de madrugada etc. mas sempre compensando as horas posteriormente (quando PJ) ou recebendo apropriadamente (quando CLT).

E esse lance de sobreposição do PSD, sinceramente acho que mandava um table-flip (metaforicamente) e iria embora. Quem pede esse tipo de coisa não tem a mínima noção do que é - ou deveria ser - desenvolvimento web.

**Comment by kazzkiq** [comment]
@fernandoguedes Concordo. Tudo é questão de comunicação. Quando o ambiente de trabalho é sadio e as equipes sabem conversar, geralmente é super fácil de contornar esses problemas de forma amigável. Mas infelizmente nem todas as empresas tem uma cultura saudável e incentivam esse tipo de diálogo.

Meu primeiro trampo foi em uma agência de pequeno porte onde o desenvolvedor-chefe lá era tão fdp que fazia questão de criticar qualquer coisa que você tivesse feito (ou que ele achasse que fosse você quem fez) em voz alta só quando o dono tava na empresa.  Debater sobre layout ou alguma mudança de escopo então, sem chance, era pedir pra ser agredido verbalmente. Por sorte, foi o único ambiente escroto em que eu trabalhei.

**Comment by tcelestino** [comment]
@LFeh, acredite!! Mas pelo o visto, teve gente que passou por coisas piores.

**Comment by bernardodiasc** [comment]
Eita, olha o "pixel perfect" causando terrores ai.. kkkkkk aprendi a liçao rapido nisso, logo que vejo inconsistencia no PSD ja do o grito. Quer codigo pixel perfect entrega design pixel perfect, ponto.

(pro tip: https://chrome.google.com/webstore/detail/perfectpixel-by-welldonec/dkaagdgjmgdmbnecmcefdhjekcoceebi?hl=en)

**Comment by dionarodrigues** [comment]
Como diz um amigo meu, "A vida é muito louca". Hehe

Quando entrei para o mundo da web ainda morava no interior, em São José do Rio Preto - que por sinal é distante de sampa, mas vem crescendo bastante. Tive sorte! Era uma empresa de TI não muito grande - tinha cerca de 15 funcionários no total - e um dos meus patrões era uma pessoa que valorizava as qualidades dos seus funcionários: no meu caso, ele sempre me motivou a estudar (deixando eu reservar 2 horas do meu dia para isso), testar tecnologias novas sempre foi muito fácil, reunir a galera para debater sobre alguns assuntos da área era comum... Assim, os problemas do dia-a-dia daquela empresa eram superados por suas vantagens.

No entanto, quando a balança entre QUALIDADES vs DEFEITOS é muito distoante, pendendo mais para os aspectos ruins, é difícil superar os problemas: o clima fica pesado, se trabalha por dinheiro... Penso, neste caso, que os pixels imperfeitos, fazer horas extras durante a madrugada e vestir a camisa da empresa fica muito complicado. Tudo tende a ir se tornando um peso sobre nossos ombros. 

---

Talvez o problema mais grave que já passei até hoje, aqui em São Paulo, foi bem parecido com o seu  @rubenmarcus : prazo x inconsistências na gestão do projeto x gestores sem noção. Neste caso, o máximo que posso fazer é explicar como meu trabalho funciona e falar a real: sugerir algumas melhorias para a organização das tarefas e prazos, dizer o tempo real (aproximado) que levarei para executar o projeto e deixar claro que estou disposto a discutir o que pode ser feito até a data da entrega combinada com o cliente - priorizar tarefas. Tem pessoas que não entende que não é possível construir uma casa inteira (já decorada) no mesmo tempo que se leva para construir apenas um quarto, tendo em vista a mesma quantidade de pessoas para executar ambas tarefas. Assim, tem patrão e gestores que gritam, sapateiam, deixam os funcionários com sensação de incompetência... Triste realidade. Claro que a correria sempre irá existir em qualquer empresa, mas a diferença está em como a cultura do lugar e às pessoas se comportarão mediante problemas como esses.

**Comment by rubenmarcus** [comment]
Ah lembrei de uma entrevista numa dessas agencias que fazem sites de Wordpress , o cara me perguntou se eu escutava rock do diabo . Ai eu perguntei porque? Ai ele: Bom, e que aqui tinha 2 desenvolvedores, que acabaram virando um casal, e eu nao gostava disso, e ele ouvia rock do diabo e levou ela pra esse caminho e me incomodava muito, e eu tive que demitir os dois sabe como e ne kkkkkkkkkkkkkkkkk Bem imbecil ele, as vezes eu tenho vontade de fazer uma lista dessas empresas

**Comment by mksanti** [comment]
rock do diado hahahahahahahaha.

**Comment by FranklinSilva** [comment]
Uma vez fui convidado para fazer uma entrevista aqui mesmo na Bahia. O rapaz ficou apenas perguntando se eu era mesmo fluente em inglês. Mas vem cá meu jovem, vc é fluente mesmo? vc entende o que as pessoas falam? "Sim, sou fluente" Mas vc sabe falar sem gaguejar? "Examente, consigo falar, ler, escrever e ouvir perfeitamente" Ahhh, pq eu queria q vc traduzisse meu site.

Dai quando eu pergunto sobre programacao ele diz "Ah, vc nao vai programar não, seu trabalho vai ser visitar clientes de onibus pela cidade e resolver os problemas deles com os nossos sistemas". Não preciso nem comentar o salário oferecido ne?

**Comment by felipefialho** [comment]
HAHAHA, não seria mais fácil contratar um tradutor?

**Comment by FranklinSilva** [comment]
HAHAHAHA Exatamente @LFeh

**Comment by clarabatt** [comment]
Trabalho em uma empresa que não existe projeto, organização e muito menos planejamento. É no "vá fazendo ai" que depois vamos ajeitando, quando chega no final já mudou tanto que nem era mais a proposta inicial. :unamused:
